contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,281 @@
+      if (groupByProperties[0]) {
+        filter = this.filterLayername_(groupByProperties[0]);
+        dataTitle = dataTitle + groupByProperties[0];
+        groupByProperties.shift();
+      } else {
+        filter = null;
+      }
+
+      this.datasets.push(this.createDataset_(datasource['url'],
+                                             filter,
+                                             dataTitle));
+    }
+    while (groupByProperties.length > 0);
+  }
+
+
+
+  /**
+   * @type {ngeox.SearchDirectiveListeners}
+   * @export
+   */
+  this.listeners = /** @type {ngeox.SearchDirectiveListeners} */ ({
+    selected: angular.bind(this, gmf.SearchController.selected_)
+  });
+
+};
+
+
+/**
+ * @return {ol.layer.Vector} The vector layer.
+ * @private
+ */
+gmf.SearchController.prototype.createVectorLayer_ = function() {
+  var vectorLayer = new ol.layer.Vector({
+    source: new ol.source.Vector()
+  });
+  // Use vectorLayer.setMap(map) rather than map.addLayer(vectorLayer). This
+  // makes the vector layer "unmanaged", meaning that it is always on top.
+  vectorLayer.setMap(this.map);
+  return vectorLayer;
+};
+
+
+/**
+ * @param {string} url A ws url that return geojson
+ * @param {Function?} filter A filter based on a featuresCollection's array
+ * @param {string} dataTitle the tile of this dataset's results
+ * @return {TypeaheadDataset} a typeahead dataset.
+ * @private
+ */
+gmf.SearchController.prototype.createDataset_ = function(url, filter,
+                                                             dataTitle) {
+  var rootScope = this.rootScope_;
+  var compile = this.compile_;
+  var bloodhoundEngine = this.createAndInitBloodhound_(url, filter);
+  return /** @type {TypeaheadDataset} */ ({
+    source: bloodhoundEngine.ttAdapter(),
+    display: function(suggestion) {
+      var feature = /** @type {ol.Feature} */ (suggestion);
+      return feature.get('label');
+    },
+    templates: {
+      header: function() {
+        return '<div class="header">' + dataTitle + '</div>';
+      },
+      suggestion: function(suggestion) {
+        var feature = /** @type {ol.Feature} */ (suggestion);
+
+        // A scope for the ng-click on the suggestion's « i » button.
+        var scope = rootScope.$new(true);
+        scope['feature'] = feature;
+        scope['click'] = function(event) {
+          window.alert(feature.get('label'));
+          event.stopPropagation();
+        };
+
+        var html = '<p>' + feature.get('label') +
+            '<button ng-click="click($event)">i</button></p>';
+        return compile(html)(scope);
+      }
+    }
+  });
+};
+
+
+/**
+ * @param {string} layername the dataset "type" to keep
+ * @return {Function} A filter function based on a featuresCollection's array
+ * @private
+ */
+gmf.SearchController.prototype.filterLayername_ = function(layername) {
+  return function(feature) {
+    return (feature['properties']['layer_name'] === layername);
```

Comment:
We should create a `gmf-search.js` externs file, with types for the objects returned in the response from the full text search. We can do that later.